qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -798,6 +798,37 @@ if [ -z "${RBD_MIRROR_USE_RBD_MIRROR}" ]; then
   CEPH_ARGS='--id admin' ceph --cluster ${CLUSTER2} osd blocklist ls 2>&1 | grep -q "listed 0 entries"
 fi
 
+if [ "${RBD_MIRROR_MODE}" = "snapshot" ]; then
+  if [ -z "${RBD_MIRROR_USE_RBD_MIRROR}" ]; then
+    testlog "TEST: test partially synced demote snapshot sync after rbd-mirror restart"
+    demote_image=test_demote_image
+    create_image_and_enable_mirror ${CLUSTER2} ${POOL} ${demote_image} ${RBD_MIRROR_MODE} 1G
+    write_image ${CLUSTER2} ${POOL} ${demote_image} 100
+    wait_for_image_replay_stopped ${CLUSTER2} ${POOL} ${demote_image}
+    wait_for_image_replay_started ${CLUSTER1} ${POOL} ${demote_image}
+    wait_for_replay_complete ${CLUSTER1} ${CLUSTER2} ${POOL} ${POOL} ${demote_image}
+    wait_for_status_in_pool_dir ${CLUSTER1} ${POOL} ${demote_image} 'up+replaying'
+    wait_for_status_in_pool_dir ${CLUSTER2} ${POOL} ${demote_image} 'up+stopped'
+    write_image ${CLUSTER2} ${POOL} ${demote_image} 256 4194304
+    demote_image ${CLUSTER2} ${POOL} ${demote_image}
+    get_newest_complete_mirror_snapshot_id ${CLUSTER2} ${POOL} ${demote_image} snap_id
+    wait_for_non_primary_snap_present ${CLUSTER1} ${POOL} ${demote_image} ${snap_id}
```

Comment:
Nit: I'd match the parameter name to make it clearer
```suggestion
    get_newest_complete_mirror_snapshot_id ${CLUSTER2} ${POOL} ${demote_image} primary_snap_id
    wait_for_non_primary_snap_present ${CLUSTER1} ${POOL} ${demote_image} ${primary_snap_id}
```